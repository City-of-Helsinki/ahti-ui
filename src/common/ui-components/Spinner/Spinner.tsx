import React from 'react';
import Lottie, { Options } from 'react-lottie';

let options: Options;

export interface SpinnerProps {
  readonly animation: typeof options.animationData;
  readonly width?: number | string;
  readonly height?: number | string;
  readonly className?: string;
}

const Spinner: React.FC<SpinnerProps> = ({
  animation,
  width = '100%',
  height = '100%',
  className,
}) => {
  const lottieOptions = {
    loop: true,
    autoplay: true,
    prerender: true,
    animationData: animation,
  };

  return (
    <div className={className}>
      <Lottie options={lottieOptions} width={width} height={height} />
    </div>
  );
};

export default Spinner;
