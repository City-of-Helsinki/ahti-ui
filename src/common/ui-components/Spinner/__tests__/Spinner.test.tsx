import React from 'react';
import { shallow } from 'enzyme';

import Spinner from '../Spinner';
import * as animation from '../animations/spinner_all.json';

describe('Spinner', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Spinner animation={animation} />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
