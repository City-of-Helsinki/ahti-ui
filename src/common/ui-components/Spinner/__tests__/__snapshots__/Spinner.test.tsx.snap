// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders without crashing 1`] = `"<div><div style=\\"width:100%;height:100%;overflow:hidden;margin:0 auto;outline:none\\" title=\\"\\" role=\\"button\\" aria-label=\\"animation\\" tabindex=\\"0\\"></div></div>"`;
